extensions/ql-vscode/src/model-editor/modeling-events.ts

```diff
@@ -6,11 +6,15 @@ import type { Method, Usage } from "./method";
 import type { ModeledMethod } from "./modeled-method";
 import type { Mode } from "./shared/mode";
 
-interface MethodsChangedEvent {
-  readonly methods: readonly Method[];
+interface ModelingStateChangedEvent {
   readonly dbUri: string;
   readonly databaseItem: DatabaseItem;
   readonly isActiveDb: boolean;
+  readonly methods?: readonly Method[];
+  readonly modeledMethods?: Readonly<Record<string, ModeledMethod[]>>;
+  readonly modifiedMethodSignatures?: ReadonlySet<string>;
+  readonly inProgressMethodSignatures?: ReadonlySet<string>;
+  readonly processedByAutoModelMethodSignatures?: ReadonlySet<string>;
```

Comment:
I think it's quite confusing that all fields are optional here, which means that the main change is that you only need to subscribe to 1 event rather than 5 different events. However, you still need to make all kinds of checks and would theoretically need to handle all combinations of fields that can be set (since none are mutually exclusive). This can still result in sending multiple webview messages when states are modified sequentially.

Reply:
What if: 
- the fields weren't optional and we always set the state
- the state is combined into a single object rather than the 5 different pieces we have

Would that cause too many unnecessary re-renders? 🤔 Are the state pieces generally related and end up changing together often? I think this is true for `modeledMethods` and `methodSignatures` at least, but unsure about the rest. 

I think `useMemo` relies on object equality so we can't just rely on that to avoid re-renders - we'd need to make sure we don't change objects, which is generally a pain IMO.